Plutarch/Api/V1.hs

```diff
@@ -574,12 +574,41 @@ instance PIsDataRepr PDCert where
 newtype PMap (k :: PType) (v :: PType) (s :: S) = PMap (Term s (PBuiltinMap k v))
   deriving (PlutusType, PIsData) via (DerivePNewtype (PMap k v) (PBuiltinMap k v))
 
--- FIXME: This fails typecheck. "Illegal type synonym family application ‘PLifted k’ in instance"
--- instance PUnsafeLiftDecl (PMap k v) where type PLifted (PMap k v) = PlutusMap.Map (PLifted k) (PLifted v)
--- deriving via
---   (DerivePConstantViaNewtype (PlutusMap.Map (PLifted k) (PLifted v)) (PMap k v) (PBuiltinMap k v))
---   instance
---     (PConstant (PlutusMap.Map (PLifted k) (PLifted v)))
+instance
+  ( Plutus.ToData (PLifted v)
+  , Plutus.ToData (PLifted k)
+  , Plutus.FromData (PLifted v)
+  , Plutus.FromData (PLifted k)
+  , PLift k
+  , PLift v
+  ) =>
+  PUnsafeLiftDecl (PMap k v)
+  where
+  type PLifted (PMap k v) = Map.Map (PLifted k) (PLifted v)
+
+instance
+  ( PLifted (PConstanted k) ~ k
+  , Plutus.ToData v
+  , Plutus.FromData v
+  , Plutus.ToData k
+  , Plutus.FromData k
+  , PConstant k
+  , PLifted (PConstanted v) ~ v
+  , Plutus.FromData v
+  , Plutus.ToData v
+  , PConstant v
+  ) =>
+  PConstant (Map.Map k v)
+  where
+  type PConstantRepr (Map.Map k v) = [(Plutus.Data, Plutus.Data)]
+  type PConstanted (Map.Map k v) = PMap (PConstanted k) (PConstanted v)
+  pconstantToRepr m = (\(x, y) -> (Plutus.toData x, Plutus.toData y)) <$> Map.toList m
+  pconstantFromRepr :: [(Plutus.Data, Plutus.Data)] -> Maybe (Map.Map k v)
+  pconstantFromRepr m = (Map.fromList <$>) $
```

Comment:
nitpick: I would argue `fmap Map.fromList $` is cleaner than `(Map.fromList <$>) $` but whatever